src/main/java/MessageListener.java

```diff
@@ -1,31 +0,0 @@
-import events.*;
-import it.sauronsoftware.cron4j.Scheduler;
-import net.dv8tion.jda.api.JDA;
-import net.dv8tion.jda.api.JDABuilder;
-import net.dv8tion.jda.api.hooks.ListenerAdapter;
-import util.RemindChecker;
-
-import javax.security.auth.login.LoginException;
-import java.util.TimerTask;
-
-public class MessageListener extends ListenerAdapter {
-    public static void main(String[] args) throws LoginException {
-        //TODO !help command
-        //TODO save remind requests to file
-        JDA api = JDABuilder.createDefault("token").build();
-        api.addEventListener(new Hello());
```

Comment:
Ogólnie jest ok, chociaż czasami chciałbyś utrzymać referencję do tego listenera, czyli na przykład mógłbyś zrobić sobie jakąś klasę Listeners w której miałbyś pola na Message, PingPong, Reminder i Help tworzone na przykład w konstruktorze.
Coś w stylu:

```
class Listeners {
private Message message;
private PingPong pingPong;
//....
Listeners() {
 this.message = new Message();
//...
}

List<Listener> getAll() {
 return Arrays.asList(message, pingPong, ...);
}
}
```

I potem w main:
```{
 Listeners listeners = new Listeners();
 for(Listener l : listeners.getAll()) {
  api.addEventListener(l);
}
}```
No chyba że możesz tam przekazać całą listę, ale nie widziałem tego w wiki JDA.

Reply:
Zaletą jest to, że w trakcie życia appki możesz później łatwiej zarządzać tymi listenerami (np.: usuwanie).